test/case_control_async/main.cpp

```diff
@@ -189,8 +187,7 @@ void greentea_teardown(const size_t passed, const size_t failed, const failure_t
     TEST_ASSERT_EQUAL(1, failed);
     TEST_ASSERT_EQUAL(REASON_CASES, failure.reason);
     TEST_ASSERT_EQUAL(LOCATION_UNKNOWN, failure.location);
-    verbose_test_teardown_handler(passed, failed, failure);
-    MBED_HOSTTEST_RESULT(true);
+    greentea_test_teardown_handler(passed, failed, failure);
```

Comment:
We are expecting a failure, which means the test succeeded. So I think this needs to be:
```cpp
verbose_test_teardown_handler(passed, failed, failure);
greentea_send_kv(TEST_ENV_TESTCASE_SUMMARY, passed, failed);
GREENTEA_TESTSUITE_RESULT(true);
```